### PR TITLE
Fix and upgrade many things

### PR DESCRIPTION
**Pull request by Attect**
see commit
it can build by Android Studio 3.4 and run on Android pie

### CODE REVIEW

File: filter/src/main/java/com/yalantis/filter/widget/Filter.kt [hunk 1]
```diff
@@ -349,15 +324,16 @@ class Filter<T : FilterModel> : FrameLayout, FilterItemListener, CollapseListene
             putBoolean(STATE_COLLAPSED, isCollapsed!!)
             val selected = mSelectedItems
             val removed = mRemovedItems
-            if (selected is Serializable) {
+            if (true) {
```
Comment (ciscorucinski) [inline filter/src/main/java/com/yalantis/filter/widget/Filter.kt]: Why not remove this line and 330?

File: filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt [hunk 2]
```diff
@@ -191,25 +188,25 @@ class FiltersListItemAnimator : SimpleItemAnimator() {
 
     override fun animateMove(holder: RecyclerView.ViewHolder, fromX: Int, fromY: Int,
                              toX: Int, toY: Int): Boolean {
-        var fromX = fromX
-        var fromY = fromY
+        var fx = fromX
```
Comment (ciscorucinski) [inline filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt]: inconsistent variable naming: Before there was `fromX` and `toX`. Now it is `fx` and `toX`. I suggest reverting back to the old naming style

File: filter/src/main/java/com/yalantis/filter/widget/Filter.kt (hunk 1 above)
Comment (Attect) [reply filter/src/main/java/com/yalantis/filter/widget/Filter.kt]: Android Studio tell me it is always true

File: filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt (hunk 2 above)
Comment (Attect) [reply filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt]: Android Studio:Name shadowed: fromX

File: filter/src/main/java/com/yalantis/filter/widget/Filter.kt (hunk 1 above)
Comment (ciscorucinski) [reply filter/src/main/java/com/yalantis/filter/widget/Filter.kt]:
```
if (true) {
    putSerializable(STATE_SELECTED, selected)
}
```
... is the same as ...

```
putSerializable(STATE_SELECTED, selected)
```

just remove the **if condition**

File: filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt (hunk 2 above)
Comment (ciscorucinski) [reply filter/src/main/java/com/yalantis/filter/animator/FiltersListItemAnimator.kt]: Ah, I overlooked the parameter names. However, that warning won't create issues in this code. You could attach `@Suppress("NAME_SHADOWING")` to those 2 variables.

Technically the old 194 - 195 code could have been named better as the `fromX` and `fromY` variables are modified in the code. Maybe something like `_fromX` and `_fromY`

Regardless, the names of `fx` and `fy` don't fit the variable naming standard and should be changed

**Comment by Attect** [comment]
update code

**Pull request closed without merging**